src/main/webapp/app/shared/date-time-picker/date-time-picker.component.ts

```diff
@@ -91,20 +92,19 @@ export class FormDateTimePickerComponent implements ControlValueAccessor {
     }
 
     /**
-     * updates the value with the passed newValue. It is checked if the field is empty or the newValue is not null. Then
-     * the passed date is valid, invalidDate can be set to false. Otherwise the date is invalid.
-     * @param newValue a valid date object or null
-     * @param inputValue the string value of #datePicker, the field is empty when inputValue == ''
+     * updates the value with the passed newValue date. The date picked using the calendar is always valid.
+     * In case the calendar was opened and no date was selected, the value does not get updated
+     * @param newValue the date picked with the date picker or '' in case the date picker was opened and save was not pressed
      */
-    updateField(newValue: any, inputValue: string) {
-        if (newValue != null || inputValue == '') {
-            this.invalidDate = false;
-        } else {
-            this.invalidDate = true;
+    updateField(newValue: any) {
+        if (newValue != '') {
```

Comment:
Can we assume that this is a string? If so, can we change the type of the parameter?

Reply:
I looked over the method again, turns out the empty string is not always correct. If the calendar was opened while an invalid input is selected and then the calender is closed without saving the date, some object (I think some kind of array gets passed). But if a date is selected in the calendar, this date then gets passed to the method. I think it makes sense to use any here.

Reply:
In that case I'd still be careful about a comparison like `newValue != ''`. JavaScript and comparison, you know...
So maybe explicitly compare to specific values you want to capture and then that should be fine.

Reply:
I changed it